### PR TITLE
Add a jax-based X-ray projector

### PR DESCRIPTION
**Pull request by Michael-T-McCann**
For now, it's the simplest (2D, not very configurable) example. Extension to 3D slice-by-slice should be straightforward.

The structure of the code requires input. I designed this so that the projector could easily be cut and pasted outside of SCICO if desired.

Timing versus ASTRA is encouraging, see `ct_projector_comparison.py`.

### CODE REVIEW

File: scico/linop/_xray.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import jax
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+
+from scico.typing import Shape
+
+from ._linop import LinearOperator
+
+
+class XRayProject(LinearOperator):
```
Comment (bwohlberg) [inline scico/linop/_xray.py:26]: Naming is not consistent with existing CT projector classes: worth discussion.

File: scico/linop/_xray.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        """
+        self._eval = projector.project
+
+        super().__init__(
+            input_shape=projector.im_shape,
+            output_shape=(len(projector.angles), *projector.det_shape),
+        )
+
+
+class ParallelFixedAxis2dProjector:
```
Comment (bwohlberg) [inline scico/linop/_xray.py:48]: Quite a few different properties squeezed into the name! If any of these may be generalized (e.g. 2d or 3d), consider moving the property out of the name and making it a parameter when the alternative is also supported.

File: scico/linop/_xray.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    """Parallel ray, single axis, 2D X-ray projector."""
+
+    def __init__(
+        self,
+        im_shape: Shape,
+        angles: ArrayLike,
+        det_length: Optional[int] = None,
+        do_dithering: bool = True,
```
Comment (bwohlberg) [inline scico/linop/_xray.py:56]: How about just `dither`?

File: scico/linop/_xray.py (hunk 1 above)
Comment (Michael-T-McCann) [reply scico/linop/_xray.py:26]: discussion: call all of them radon transform, clarify in the docs

Comment (Michael-T-McCann) [reply scico/linop/_xray.py:26]: New discussion: `XRayTransform`

**Pull request merged**